### PR TITLE
Delete Modal

### PR DESCRIPTION
**Pull request by jasonmadigan**
Implements https://github.com/Kuadrant/console-plugin/issues/16

- Added a delete action to the DropdownWithKebab
- Used some componentry from `@patternfly/react-core/next` for a new Modal
- Modal has a confirm action before delete
- Made some tweaks to `AllPoliciesListPage` - combined the `useK8sWatchResource` hooks into a single array mapping for sanity

Testing:

Create a test resource, for example a TLSPolicy:

```bash
kubectl apply -f - <<EOF
apiVersion: gateway.networking.k8s.io/v1
kind: Gateway
metadata:
  name: jm-hypergateway
  namespace: jmadigan-test
spec:
  gatewayClassName: istio
  listeners:
  - allowedRoutes:
      namespaces:
        from: All
    name: api
    hostname: "*.jm.hcpapps.net"
    port: 443
    protocol: HTTPS
    tls:
      mode: Terminate
      certificateRefs:
        - name: apps-hcpapps-tls
          kind: Secret
EOF


kubectl apply -f - <<EOF
apiVersion: kuadrant.io/v1alpha1
kind: TLSPolicy
metadata:
  name: jm-hypergateway-tls-policy
  namespace: jmadigan-test
spec:
  targetRef:
    name: jm-hypergateway
    group: gateway.networking.k8s.io
    kind: Gateway
  issuerRef:
    group: cert-manager.io
    kind: ClusterIssuer
    name: not-found
EOF
```

### CODE REVIEW

File: src/components/KuadrantPoliciesPage.tsx [hunk 1]
```diff
@@ -394,7 +445,7 @@ const KuadrantPoliciesPage: React.FC = () => {
         href: 'tls',
         name: 'TLS',
         component: TLS
```
Comment (jasonmadigan) [inline src/components/KuadrantPoliciesPage.tsx:447]: nit: add {{ t }}

File: src/components/KuadrantPoliciesPage.tsx [hunk 2]
```diff
@@ -129,58 +167,112 @@ const AllPoliciesTable: React.FC<AllPoliciesTableProps> = ({ data, unfilteredDat
...
+
+      <Modal
+        isOpen={isDeleteModalOpen}
+        onClose={() => setIsDeleteModalOpen(false)}
+        aria-labelledby="delete-modal-title"
+        aria-describedby="delete-modal-body"
+      >
+        <ModalHeader title="Confirm Delete" />
+        <ModalBody>
+          Are you sure you want to delete the resource {obj.metadata.name}?
```
Comment (jasonmadigan) [inline src/components/KuadrantPoliciesPage.tsx:261]: add {{ t }}

File: src/components/KuadrantPoliciesPage.tsx [hunk 3]
```diff
@@ -129,58 +167,112 @@ const AllPoliciesTable: React.FC<AllPoliciesTableProps> = ({ data, unfilteredDat
...
+          </MenuToggle>
+        )}
+        shouldFocusToggleOnSelect
+      >
+        <DropdownList>
+          <DropdownItem value="edit" key="edit">
+            Edit
+          </DropdownItem>
+          <DropdownItem value="delete" key="delete">
+            Delete
```
Comment (jasonmadigan) [inline src/components/KuadrantPoliciesPage.tsx:248]: add {{ t }}

File: src/components/KuadrantPoliciesPage.tsx [hunk 4]
```diff
@@ -129,58 +167,112 @@ const AllPoliciesTable: React.FC<AllPoliciesTableProps> = ({ data, unfilteredDat
...
+            isExpanded={isOpen}
+          >
+            <EllipsisVIcon />
+          </MenuToggle>
+        )}
+        shouldFocusToggleOnSelect
+      >
+        <DropdownList>
+          <DropdownItem value="edit" key="edit">
+            Edit
```
Comment (jasonmadigan) [inline src/components/KuadrantPoliciesPage.tsx:245]: add {{ t }}

**Comment by david-martin** [comment]
👍 

Would be best to limit width of modal
image

**Comment by jasonmadigan** [comment]
@david-martin good spot, done. Added some i18n strings too.

File: src/components/KuadrantPoliciesPage.tsx [hunk 5]
```diff
@@ -129,58 +167,114 @@ const AllPoliciesTable: React.FC<AllPoliciesTableProps> = ({ data, unfilteredDat
...
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = React.useState(false);
 
   const onToggleClick = () => {
     setIsOpen(!isOpen);
   };
 
-  const onSelect = (_event: React.MouseEvent<Element, MouseEvent> | undefined, value: string | number | undefined) => {
-    // eslint-disable-next-line no-console
-    console.log('selected', value);
+  const getModelFromResource = (obj: K8sResourceCommon): K8sModel => {
```
Comment (david-martin) [inline src/components/KuadrantPoliciesPage.tsx:179]: I'll likely move this out for reuse after merge.

**Review by david-martin — APPROVED** [review]

**Pull request merged**